sklearn/tree/tree.py

```diff
@@ -282,10 +300,7 @@ def fit(self, X, y, sample_weight=None, check_input=True,
         else:
             min_weight_leaf = 0.
 
-        # Set min_samples_split sensibly
-        min_samples_split = max(self.min_samples_split,
-                                2 * self.min_samples_leaf)
-
+        min_samples_split = max(min_samples_split, 2 * self.min_samples_leaf)
```

Comment:
shouldnt you compare with `min_samples_leaf` (and not `self.min_samples_leaf`), in case the latter was float?